packages/dds/tree/src/test/simple-tree/core/allowedTypes.spec.ts

```diff
@@ -0,0 +1,480 @@
+	type I13 = InsertableTreeNodeFromImplicitAllowedTypes<typeof booleanSchema>;
+	type _check13 = requireTrue<areSafelyAssignable<I13, boolean>>;
+}
+
+// InsertableTreeNodeFromAllowedTypes
+{
+	class A extends schema.object("A", { x: [schema.number, schema.string] }) {}
+	class B extends schema.object("B", { x: [schema.number, schema.null] }) {}
+
+	{
+		type T = InsertableTreeNodeFromAllowedTypes<AllowedTypes>;
+		type _check = requireAssignableTo<T, never>;
+	}
+
+	{
+		type T = InsertableTreeNodeFromAllowedTypes<[typeof numberSchema, typeof stringSchema]>;
+		type _check = requireTrue<areSafelyAssignable<T, number | string>>;
+	}
+
+	{
+		type T = InsertableTreeNodeFromAllowedTypes<[]>;
+		type _check = requireAssignableTo<T, never>;
+	}
+
+	{
+		type T = InsertableTreeNodeFromAllowedTypes<[typeof A, typeof B]>;
+		type _check = requireAssignableTo<A | B, T>;
+	}
+
+	{
+		type T = InsertableTreeNodeFromAllowedTypes<[typeof A, typeof B] | [typeof A]>;
+		type _check = requireAssignableTo<T, never>;
+	}
+
+	{
+		type T = InsertableTreeNodeFromAllowedTypes<[typeof B, typeof A] | [typeof A]>;
+		type _check = requireAssignableTo<T, never>;
+	}
+
+	{
+		type T = InsertableTreeNodeFromAllowedTypes<[typeof A, typeof B | typeof A]>;
+		type _check = requireAssignableTo<A, T>;
+		type _check2 = requireFalse<isAssignableTo<B, T>>;
+	}
+}
+
+// Type tests for unannotate utilities
+{
+	// UnannotateImplicitAllowedTypes
+	{
+		{
+			type _check = requireAssignableTo<
+				UnannotateImplicitAllowedTypes<ImplicitAnnotatedAllowedTypes>,
+				ImplicitAllowedTypes
+			>;
+		}
+
+		{
+			type T = TreeNodeSchema;
+			type _check = requireAssignableTo<
+				T,
+				UnannotateImplicitAllowedTypes<ImplicitAnnotatedAllowedTypes>
+			>;
+		}
+
+		{
+			type T = AllowedTypes;
+			type _check = requireAssignableTo<
+				T,
+				UnannotateImplicitAllowedTypes<ImplicitAnnotatedAllowedTypes>
+			>;
+		}
+	}
+
+	// UnannotateAllowedTypeOrLazyItem
+	{
+		type A = LazyItem<TreeNodeSchema>;
+		type B = AnnotatedAllowedType;
+
+		type _check1 = requireAssignableTo<A, UnannotateAllowedTypeOrLazyItem<A>>;
+
+		type _check2 = requireAssignableTo<UnannotateAllowedTypeOrLazyItem<A>, A>;
+
+		type _check3 = requireAssignableTo<UnannotateAllowedTypeOrLazyItem<B>, A>;
+	}
+
+	// UnannotateAllowedTypesList
+	{
+		type A1 = AnnotatedAllowedType;
+		type A2 = LazyItem<TreeNodeSchema>;
+		type Mixed = readonly [A1, A2];
+
+		type Empty = readonly [];
+		type _check1 = requireAssignableTo<Empty, UnannotateAllowedTypesList<Empty>>;
+
+		type _check2 = requireAssignableTo<UnannotateAllowedTypesList<Mixed>, readonly A2[]>;
+	}
+
+	// UnannotateAllowedTypes
+	{
+		type AnnotatedList = readonly [AnnotatedAllowedType, LazyItem<TreeNodeSchema>];
+
+		type _check = requireAssignableTo<
+			UnannotateAllowedTypes<{
+				metadata: AllowedTypesMetadata;
+				types: AnnotatedList;
+			}>,
+			readonly [LazyItem<TreeNodeSchema>, LazyItem<TreeNodeSchema>]
+		>;
+	}
+}
+
+describe("allowedTypes", () => {
```

Comment:
```suggestion
describe("allowed type utilities", () => {
```

Reply:
I prefer to name the top level describe block after the file it is testing (and thus also the name of the file it is in), in this case allowedTypes. Since our test browser tools don't provide any way to navigate by file, keeping these aligned makes it easier to find tests and saves us from having to invest a second name for every file.